### PR TITLE
Add introductory COS notebook (Running CalCOS Pipeline Notebook)

### PR DESCRIPTION
**Pull request by nkerman**
Going branch by branch of the new COS notebooks and PR-ing them to the spacetelescope/notebooks master repo.
This one is the CalCOS subdir.

A ton of commits (more than 255) went into these notebooks, but to keep the git history small, I am only PR-ing with the latest (as of May 13 2021) versions. The original histories should be found at my fork (nkerman/notebooks/)

### CODE REVIEW

File: notebooks/COS/CalCOS/CalCOS.ipynb [hunk 1]
```diff
@@ -0,0 +1,931 @@
...
+    "              outdir=output_dir+\"/calcos_processed_1\") # save all resulting files in this subdirectory in our output directory"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open(output_dir+'output_calcos_1.txt', 'w') as f: # This file now contains the output of the last cell\n",
```
Comment (eteq) [inline notebooks/COS/CalCOS/CalCOS.ipynb:541]:
```suggestion
    "with open(output_dir + 'output_calcos_1.txt', 'w') as f: # This file now contains the output of the last cell\n",
```
Another minor suggestion

**Review by eteq — CHANGES_REQUESTED** [review]
Mostly good aside from a few minor style quibbles and clarity suggestions! Main concern is the desire to switch to paths that don't depend on shell commands

File: notebooks/COS/CalCOS/CalCOS.ipynb [hunk 2]
```diff
@@ -0,0 +1,931 @@
...
+    "To determine which reference files were used most recently by STScI to calibrate your data (often, but not always the newest and best,) you can refer to your data file's \"CRDS_CTX\" keyword in its fits header (see next cell)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rawfiles = glob.glob('data/*raw*.fits') # find all of the raw files\n",
```
[the remark this answers is not in the document]
Comment (nkerman) [reply notebooks/COS/CalCOS/CalCOS.ipynb:331]: For now I think I will just call str on all the Path objects.

File: notebooks/COS/CalCOS/CalCOS.ipynb [hunk 3]
```diff
@@ -0,0 +1,931 @@
...
+    "<a id = prereqC></a>\n",
+    "## 1.1. Prerequisites\n",
+    "This tutorial assumes some basic knowledge of the command line and was built using a unix bash-style shell. Those using a Windows computer will likely have the best results if working within the [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/install-win10).\n",
+    "\n",
+    "\n",
+    "If you do not already have any distribution of the `conda` tool, see [this page](https://astroconda.readthedocs.io/en/latest/getting_started.html#getting-started-jump) for instructions, and install either [`anaconda` - more beginner friendly, \\~ 3 GB, lots of extras you likely won't use](https://docs.anaconda.com/anaconda/install/) or [`miniconda` - \\~ 400 MB, only what you need](https://docs.conda.io/en/latest/miniconda.html).\n",
+    "\n",
+    "<a id = condaenvC></a>\n",
+    "## 1.2. Create your conda environment\n",
+    "Once you have `conda` installed, it's time to create an environment. If you already setup an `astroconda` environment in the notebook on \"Setting up your environment\", you may skip this and merely activate the environment you created then.\n",
```
[the remark this answers is not in the document]
Comment (nkerman) [reply notebooks/COS/CalCOS/CalCOS.ipynb:152]: I see this notebook as perhaps the most likely one that people will just go straight to without looking at the other notebooks. I'm going to mention the Setup, and give a hypertext link to skip the section if they've done it.

**Review by eteq — APPROVED** [review]

**Pull request merged**